test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -1824,14 +1825,22 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 		})
 
 		AfterEach(func() {
-			//Revert the profile
-			profiles.UpdateWithRetry(initialProfile)
+			profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
 
-			By("Applying changes in performance profile and waiting until mcp will start updating")
-			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+			currentSpec, _ := json.Marshal(profile.Spec)
+			spec, _ := json.Marshal(initialProfile.Spec)
+			// revert only if the profile changes.
+			if !bytes.Equal(currentSpec, spec) {
```

Comment:
nice idea. Reflect.DeepEqual only on the `Spec` should have worked as well. No need to change it now -unless this approach doesn't work.

Reply:
If you resubmit: bail out earlier and log why. IOW:
```
if bytes.Equal(currentSpec, spec) {
    framework.Logf("no profile change, avoiding revert") // or klog.Infof, whatever we use
    return
}

// revert profile
...
```

Reply:
@yanirq I told ya it's better keep the happy path on the left :)